### PR TITLE
 🐛 fix webhooks

### PR DESCRIPTION
**Pull request by alexeldeib**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/608

**Special notes for your reviewer**:

**Release note**:

```release-note
Cluster and machine controllers correctly enforce webhook validation.
```

### CODE REVIEW

File: api/v1alpha3/azurecluster_webhook.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ func (c *AzureCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		Complete()
 }
 
-// +kubebuilder:webhook:verbs=create;update;delete,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-azurecluster,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=azurecluster,versions=v1alpha3,name=validation.azurecluster.infrastructure.cluster.x-k8s.io,sideEffects=None
+// +kubebuilder:webhook:verbs=create;update;delete,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-azurecluster,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=azureclusters,versions=v1alpha3,name=validation.azurecluster.infrastructure.cluster.x-k8s.io,sideEffects=None
```
Comment (alexeldeib) [inline api/v1alpha3/azurecluster_webhook.go:35]: one

File: api/v1alpha3/azuremachine_webhook.go [hunk 2]
```diff
@@ -35,7 +35,7 @@ func (m *AzureMachine) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		Complete()
 }
 
-// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-azuremachine,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=azuremachine,versions=v1alpha3,name=validation.azuremachine.infrastructure.cluster.x-k8s.io,sideEffects=None
+// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-azuremachine,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=azuremachines,versions=v1alpha3,name=validation.azuremachine.infrastructure.cluster.x-k8s.io,sideEffects=None
```
Comment (alexeldeib) [inline api/v1alpha3/azuremachine_webhook.go:38]: two

**Comment by CecileRobertMichon** [comment]
/approve
/lgtm

good find @alexeldeib 👏

**Comment by alexeldeib** [comment]
me after realizing this:
we did it

us on friday **not** debugging this:
party

**Comment by alexeldeib** [comment]
@CecileRobertMichon the treasure is in the celebtratory gifs

**Comment by devigned** [comment]
This type of failure is a foot gun. This should be something that we should be able to catch through linting or something.

**Pull request merged**